debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/WalPositionLocator.java

```diff
@@ -71,6 +71,12 @@ public Optional<Lsn> resumeFromLsn(Lsn currentLsn, ReplicationMessage message) {
             // We can resume streaming from it
             if (currentLsn.equals(lastEventStoredLsn)) {
                 // BEGIN and first message after change have the same LSN
+                if (txStartLsn != null) {
+                    // start from the BEGIN tx; prevent skipping of unprocessed event after BEGIN
```

Comment:
@rajdangwal Could we avoid the reprocessing if we add the message type to offsets? In that case we will see if it was tx begin or change and in that case we could choose the right branch.
Also for upgrade if the message type is not in offsets then choose the safe path causing potential reprocessing.
WDYT?

Reply:
Thanks @jpechane 
Sounds great. I will work on the same.

Reply:
Made the changes, but I am not able to add the respective tests because can't find a way to restart the connector after committing only the BEGIN message.